app/config/AppConfig.scala

```diff
@@ -131,12 +131,14 @@ class AppConfigImpl @Inject() (config: ServicesConfig, configuration: Configurat
   def apiVersionReleasedInProduction(version: String): Boolean = config.getBoolean(s"api.$version.endpoints.api-released-in-production")
 
   def endpointReleasedInProduction(version: String, name: String): Boolean = {
-    val switches = endpointSwitches(version)
-    switches.getOrElse(name, apiVersionReleasedInProduction(version))
+    val endpointEnabledMap = getEndpointEnabledMap(version)
+    val releasedInProd     = apiVersionReleasedInProduction(version)
+
+    if (releasedInProd) endpointEnabledMap.getOrElse(name, releasedInProd) else releasedInProd
   }
 
-  private def endpointSwitches(version: String): Map[String, Boolean] = {
-    val path = s"api.$version.endpoints.switches"
+  private def getEndpointEnabledMap(version: String): Map[String, Boolean] = {
```

Comment:
bit confused by the method name, as it's returning the "released in production" config.

(Also, "get" is a bit JavaBeans-ey, (non-functional), and it shouldn't really have "Map" in the name i.e. specifying the type)